services/explorer/src/pages/BlockData/DAGDataContainer.effects.tsx

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useQuery } from 'urql';
+import { dagDetailsQuery } from '../../api';
+import { useExplorerNetwork } from '../../hooks/useExplorerNetwork';
+import { useExplorerLoader } from '../../hooks/useLoader';
+import { DagBlock, Transaction } from '../../models';
+
+export const useDAGDataContainerEffects = (hash: string) => {
+  const { currentNetwork } = useExplorerNetwork();
+  const [blockData, setBlockData] = useState<DagBlock>({} as DagBlock);
+  const [transactions, setTransactions] = useState<Transaction[]>([
+    {} as Transaction,
+  ]);
+  const [{ fetching, data }] = useQuery({
+    query: dagDetailsQuery,
+    variables: {
+      hash,
+    },
+    pause: !hash,
+  });
+  const { initLoading, finishLoading } = useExplorerLoader();
+
+  const fetchBlockDetails = useCallback(() => {
+    setTimeout(() => {
```

Comment:
You don't need the setTimeout anymore here. And this can be wrapped inside an useEffect like in the other pages.